Coq 8.15.1 doesn't build under Cygwin

Running `opam install coq=8.15.1` under Cygwin aborts at the following point:
```
# DUNE      sources
# Error: Don't know how to build _build/install/default/bin/coqdep.exe
# DUNE      _build/install/default/bin/coqc.exe
# DUNE      _build/install/default/bin/coqproofworker.opt.exe
# Error: Don't know how to build _build/install/default/bin/coqc.exe
# make[1]: *** [Makefile.common:131: _build/install/default/bin/coqc.exe] Error 1
# make[1]: *** Waiting for unfinished jobs....
# Error: Don't know how to build
# _build/install/default/bin/coqproofworker.opt.exe
# make[1]: *** [Makefile.common:131: _build/install/default/bin/coqproofworker.opt.exe] Error 1
```
Coq 8.13.2 installs fine on the same platform.

And the full output of `make`:
```
make --warn-undefined-variable --no-builtin-rules -f Makefile.build 
make[1]: Entering directory '/home/ci/coq-8.15.1'
MKDIR     BUILD_OUT
DUNE      _build/install/default/bin/coqdep.exe
DUNE      sources
Error: Don't know how to build _build/install/default/bin/coqdep.exe
DUNE      _build/install/default/bin/coqc.exe
DUNE      _build/install/default/bin/coqproofworker.opt.exe
Error: Don't know how to build _build/install/default/bin/coqc.exe
make[1]: *** [Makefile.common:131: _build/install/default/bin/coqc.exe] Error 1
make[1]: *** Waiting for unfinished jobs....
Error: Don't know how to build
_build/install/default/bin/coqproofworker.opt.exe
make[1]: *** [Makefile.common:131: _build/install/default/bin/coqproofworker.opt.exe] Error 1
make[1]: Leaving directory '/home/ci/coq-8.15.1'
make: *** [Makefile.make:122: submake] Error 2
```

--- a/tools/configure/configure.ml
+++ b/tools/configure/configure.ml
@@ -42,10 +42,12 @@ let coq_strict_sequence = "-strict-sequence"
 let arch prefs = arch prefs.arch
 
 (** NB: [arch_is_win32] is broader than [os_type_win32], cf. cygwin *)
-
 let arch_is_win32 arch = arch = "win32"
 
-let resolve_binary_suffix arch = if arch_is_win32 arch then ".exe" else ""
+let unix arch = os_type_cygwin || not (arch_is_win32 arch)
+
+(* For cygwin dune doesn't add .exe to the binaries, only under mingw *)
+let resolve_binary_suffix arch = if unix arch then "" else ".exe"
 
 (** * Git Precommit Hook *)
 let install_precommit_hook prefs =
@@ -238,8 +240,6 @@ let check_doc () =
 
 (* Source code root *)
 let coqsrc = Sys.getcwd ()
-
-let unix arch = os_type_cygwin || not (arch_is_win32 arch)
 
 (** Variable name, description, ref in prefs, default dir, prefix-relative *)
 

